README.md

```diff
@@ -0,0 +1,273 @@
+# Context Chat
+> [!WARNING]
+> This is a beta software. Expect breaking changes.
+
+> [INFO]
+> A fully working manually example with cuda 11.8 is at the end of this readme
+
+## Simple Install
+1. Install three mandatory apps for this app to work as desired in your Nextcloud install from the "Apps" page:
+- AppAPI (>= v2.0.3): https://apps.nextcloud.com/apps/app_api
+- Context Chat (>= 1.1.0): https://apps.nextcloud.com/apps/context_chat
+- Assistant: https://apps.nextcloud.com/apps/assistant (The OCS API or the `occ` commands can also be used to interact with this app but it recommended to do that through a Text Processing OCP API consumer like the Assitant app.)
+2. Install this backend app (Context Chat Backend: https://apps.nextcloud.com/apps/context_chat_backend) from the "External Apps" page
+3. Start using Context Chat from the Assistant UI
+
+> [!NOTE]
+> See [AppAPI's deploy daemon configuration](#configure-the-appapis-deploy-daemon)
+> For GPU Support:
+> Ensure docker is installed and the Nextcloud's web server user has access to `/var/run/docker.sock`, the docker socket.
+> Mount the docker.sock in the Nextcloud container if you happen to use a containerized install of Nextcloud and ensure correct permissions for the web server user to access it.
+> See 4th point in [Complex Install (with docker)](#complex-install-with-docker) on how to do this
+
+## Complex Install (without docker)
+
+1.  `python -m venv .venv`
+2.  `. .venv/bin/activate`
+3. Install requirements
+
+    3.1 For using CPU:
+    
+    `pip install --no-deps -r requirements.cpu.txt`
+    
+    3.2 Using GPU with CUDA:
+    
+    `pip install --no-deps -r requirements.cuda.txt`
+4. Install pandoc from your desired package manager (`# apt install pandoc` for Debian-based systems)
+5. Copy example.env to .env and fill in the variables
+6. Configure `config.cpu.yaml`  *(or **config.cuda.yaml** using **gpu** and replace config.cpu.yaml with it)* for the model name, model type and its parameters (which also includes model file's path and model id as per requirements, see example config)
+7.  `./main.py`
+8. [Follow the below steps to register the app in the app ecosystem](#register-as-an-ex-app)
+
+## Complex Install (with docker)
+1. Build the image
+    *(this is a good place to edit the example.env file before building the container)*
+
+    1.1 **CPU:**
+    
+    `docker build -t context_chat_backend_dev . -f Dockerfile.cpu`
+    
+    1.2 **GPU:**
+    
+    `docker build -t context_chat_backend_dev . -f Dockerfile.cuda11.8`
+
+2.  `docker run --add-host=host.docker.internal:host-gateway -p 10034:10034 context_chat_backend_dev`
+3. Volumes can be mounted for `model_files` and `vector_db_files` if you wish with `-v $(pwd)/model_files:/app/model_files` and similar for vector_db_files
```

Comment:
I forgot to update this :p. Since we are already here, it would be nice to change `model_files` to `persistent_storage/model_files` and same for the `vector_db_files`